File: SDKs/googleapis.com/integrations/v1alpha/python/src/sdk/models/shared/enterprisecrmfrontendseventbusprotoparamspecentry.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *

class EnterpriseCrmFrontendsEventbusProtoParamSpecEntryDataTypeEnum(str, Enum):
    DATA_TYPE_UNSPECIFIED = "DATA_TYPE_UNSPECIFIED"
    STRING_VALUE = "STRING_VALUE"
    INT_VALUE = "INT_VALUE"
    DOUBLE_VALUE = "DOUBLE_VALUE"
    BOOLEAN_VALUE = "BOOLEAN_VALUE"
    PROTO_VALUE = "PROTO_VALUE"
    SERIALIZED_OBJECT_VALUE = "SERIALIZED_OBJECT_VALUE"
    STRING_ARRAY = "STRING_ARRAY"
    INT_ARRAY = "INT_ARRAY"
    DOUBLE_ARRAY = "DOUBLE_ARRAY"
    PROTO_ARRAY = "PROTO_ARRAY"
    PROTO_ENUM = "PROTO_ENUM"
    BOOLEAN_ARRAY = "BOOLEAN_ARRAY"
    PROTO_ENUM_ARRAY = "PROTO_ENUM_ARRAY"
    BYTES = "BYTES"
    BYTES_ARRAY = "BYTES_ARRAY"
    NON_SERIALIZABLE_OBJECT = "NON_SERIALIZABLE_OBJECT"
    JSON_VALUE = "JSON_VALUE"


@dataclass_json
@dataclass
class EnterpriseCrmFrontendsEventbusProtoParamSpecEntry:
    r"""EnterpriseCrmFrontendsEventbusProtoParamSpecEntry
    Key-value pair of EventBus task parameters. Next id: 13
    """
    
    class_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('className') }})
    collection_element_class_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('collectionElementClassName') }})
    config: Optional[EnterpriseCrmEventbusProtoParamSpecEntryConfig] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('config') }})
    data_type: Optional[EnterpriseCrmFrontendsEventbusProtoParamSpecEntryDataTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataType') }})
    default_value: Optional[EnterpriseCrmFrontendsEventbusProtoParameterValueType] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('defaultValue') }})
    is_deprecated: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('isDeprecated') }})
    is_output: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('isOutput') }})
    json_schema: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('jsonSchema') }})
    key: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('key') }})
    proto_def: Optional[EnterpriseCrmEventbusProtoParamSpecEntryProtoDefinition] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('protoDef') }})
    required: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('required') }})
    validation_rule: Optional[EnterpriseCrmEventbusProtoParamSpecEntryValidationRule] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('validationRule') }})
    

File: SDKs/googleapis.com/integrations/v1alpha/typescript/src/sdk/models/shared/enterprisecrmfrontendseventbusprotoparamspecentry.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { EnterpriseCrmEventbusProtoParamSpecEntryConfig } from "./enterprisecrmeventbusprotoparamspecentryconfig";
import { EnterpriseCrmFrontendsEventbusProtoParameterValueType } from "./enterprisecrmfrontendseventbusprotoparametervaluetype";
import { EnterpriseCrmEventbusProtoParamSpecEntryProtoDefinition } from "./enterprisecrmeventbusprotoparamspecentryprotodefinition";
import { EnterpriseCrmEventbusProtoParamSpecEntryValidationRule } from "./enterprisecrmeventbusprotoparamspecentryvalidationrule";


export enum EnterpriseCrmFrontendsEventbusProtoParamSpecEntryDataTypeEnum {
    DataTypeUnspecified = "DATA_TYPE_UNSPECIFIED",
    StringValue = "STRING_VALUE",
    IntValue = "INT_VALUE",
    DoubleValue = "DOUBLE_VALUE",
    BooleanValue = "BOOLEAN_VALUE",
    ProtoValue = "PROTO_VALUE",
    SerializedObjectValue = "SERIALIZED_OBJECT_VALUE",
    StringArray = "STRING_ARRAY",
    IntArray = "INT_ARRAY",
    DoubleArray = "DOUBLE_ARRAY",
    ProtoArray = "PROTO_ARRAY",
    ProtoEnum = "PROTO_ENUM",
    BooleanArray = "BOOLEAN_ARRAY",
    ProtoEnumArray = "PROTO_ENUM_ARRAY",
    Bytes = "BYTES",
    BytesArray = "BYTES_ARRAY",
    NonSerializableObject = "NON_SERIALIZABLE_OBJECT",
    JsonValue = "JSON_VALUE"
}


// EnterpriseCrmFrontendsEventbusProtoParamSpecEntry
/** 
 * Key-value pair of EventBus task parameters. Next id: 13
**/
export class EnterpriseCrmFrontendsEventbusProtoParamSpecEntry extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=className" })
  className?: string;

  @SpeakeasyMetadata({ data: "json, name=collectionElementClassName" })
  collectionElementClassName?: string;

  @SpeakeasyMetadata({ data: "json, name=config" })
  config?: EnterpriseCrmEventbusProtoParamSpecEntryConfig;

  @SpeakeasyMetadata({ data: "json, name=dataType" })
  dataType?: EnterpriseCrmFrontendsEventbusProtoParamSpecEntryDataTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=defaultValue" })
  defaultValue?: EnterpriseCrmFrontendsEventbusProtoParameterValueType;

  @SpeakeasyMetadata({ data: "json, name=isDeprecated" })
  isDeprecated?: boolean;

  @SpeakeasyMetadata({ data: "json, name=isOutput" })
  isOutput?: boolean;

  @SpeakeasyMetadata({ data: "json, name=jsonSchema" })
  jsonSchema?: string;

  @SpeakeasyMetadata({ data: "json, name=key" })
  key?: string;

  @SpeakeasyMetadata({ data: "json, name=protoDef" })
  protoDef?: EnterpriseCrmEventbusProtoParamSpecEntryProtoDefinition;

  @SpeakeasyMetadata({ data: "json, name=required" })
  required?: boolean;

  @SpeakeasyMetadata({ data: "json, name=validationRule" })
  validationRule?: EnterpriseCrmEventbusProtoParamSpecEntryValidationRule;
}
